content/components/selectpanel.mdx

```diff
@@ -64,58 +168,329 @@ Select panel should be anchored to its trigger.
+      eliminating the need for declarative buttons.
+    </Caption>
+  </Do>
+  <Dont>
+    <img
+      width="464"
+      alt="Mouse cursor indicating a click action outside of the dialog."
+      src="https://github.com/primer/design/assets/980622/ed1c44fb-b82a-4388-8ebb-542b413bdb36"
+    />
+    <Caption>Avoid combining declarative buttons with single select items that feature checkmark icons.</Caption>
+  </Dont>
+</DoDontContainer>
+
+When an additional checkbox is added to the footer, declarative buttons are required, and the items should be accompanied by visually resembling radio buttons. Although they are not technically radio buttons in terms of semantics, using visual radio buttons can effectively convey that the selection process is not yet complete even after making a choice.
+
+<DoDontContainer>
+  <Do>
+    <img
+      width="464"
+      alt="Mouse cursor indicating a clicking action on the apply button."
+      src="https://github.com/primer/design/assets/980622/edb97706-38c1-4464-bb1e-0f9ce5c41d7e"
+    />
+    <Caption>
+      When adding an additional checkbox in the footer, it is necessary to use declarative buttons and ensure that the
+      items are accompanied by radio buttons.
+    </Caption>
+  </Do>
+  <Dont>
+    <img
+      width="464"
+      alt="Mouse cursor indicating a click action outside of the dialog."
+      src="https://github.com/primer/design/assets/980622/053719a2-ef42-4078-9cf4-4f3b756e5218"
+    />
+    <Caption>Avoid mixing declaritive buttons and items with checkmark icons.</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Clearing
+
+Within the select panel, there are multiple ways to clear data. This can involve either clearing the search/filter input or clearing the current selection. It is crucial to avoid confusing their functionality, as one is related to the list itself, while the other pertains solely to the search/filter input.
+
+<DoDontContainer>
+  <Do>
+    <img
+      width="464"
+      alt="Mouse cursor indicating a clicking action on the apply button."
+      src="https://github.com/primer/design/assets/980622/4e11eb8a-c515-49c8-9b48-75330b72a963"
+    />
+    <Caption>Display a clear button to remove the input value when it is not empty.</Caption>
+  </Do>
+  <Dont>
+    <img
+      width="464"
+      alt="Mouse cursor indicating a click action outside of the dialog."
+      src="https://github.com/primer/design/assets/980622/079feed7-4ddc-476a-b639-ea17de141b9d"
+    />
+    <Caption>Don't forget to show a clear button when the input value is not empty.</Caption>
+  </Dont>
+</DoDontContainer>
+
+The clear selection [icon button](/components/icon-button) is displayed as soon as at least one item is selected and disappears once the selection is cleared.
+
+<DoDontContainer>
+  <Do>
+    <img
+      width="464"
+      alt="Mouse cursor indicating a clicking action on the apply button."
+      src="https://github.com/primer/design/assets/980622/381b2224-3ad0-4320-94cf-8284eb3f5157"
```

Comment:
I read this as a "clear filters" icon, not "deselect all selections" icon. I have _no_ idea what that icon might look like 😅

Reply:
Same 😅

Reply:
I've rewritten this section. Let me know what you think.